### PR TITLE
Not existant constant; fixes #1592

### LINKED ISSUE #1592: [GLPI 9.1.1] TicketValidation::CREATEDEMAND is declared nowhere!!!
Hello,

If you try to search for ticket on Ticket Approval - Requester field on ticket.php you get an error message:

> Uncaught Error: Undefined class constant 'CREATEDEMAND' in C:\inetpub\wwwroot\glpi091\inc\user.class.php:3032
> 
> Stack trace:
> 
> #0 C:\inetpub\wwwroot\glpi091\ajax\getDropdownUsers.php(94): User::getSqlSearchResult(false, Array, Array, 0, Array, '', 0, 200)
> 
> #1 {main}

I searched in the complete GLPI tree, and I cannot find this constant.

Regards,
Tomolimo

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

File: inc/user.class.php [hunk 1]
```diff
@@ -3020,8 +3020,7 @@ static function getSqlSearchResult ($count=true, $right="all", $entity_restrict=
 
                   case 'create_ticket_validate' :
                      $where[]= " (`glpi_profilerights`.`name` = 'ticketvalidation'
-                                  AND (`glpi_profilerights`.`rights` & ".TicketValidation::CREATEREQUEST."
-                                       OR `glpi_profilerights`.`rights` & ".TicketValidation::CREATEDEMAND.") ".
```
Comment (orthagh) [inline inc/user.class.php]: Imo, we should replacing TicketValidation::CREATEDEMAND by TicketValidation::CREATEINCIDENT
instead removing line

Comment (trasher) [inline inc/user.class.php]: @tomolimo?

Comment (tomolimo) [inline inc/user.class.php]: Let's start with both rights: it seems to be consistent with the rest of the code

```
case 'create_ticket_validate' :
                     $where[]= " (`glpi_profilerights`.`name` = 'ticketvalidation'
                                  AND (`glpi_profilerights`.`rights` & ".TicketValidation::CREATEREQUEST."
                                       OR `glpi_profilerights`.`rights` & ".TicketValidation::CREATEINCIDENT.") ".
                                  getEntitiesRestrictRequest("AND", "glpi_profiles_users", '',
                                                             $entity_restrict, 1).") ";
                     $forcecentral = false;
                     break;
```

Comment (trasher) [inline inc/user.class.php]: OK, thank you :) I've done the modification

**Pull request merged**